### PR TITLE
Sapflow QA/QC

### PR DESCRIPTION
**Pull request by stephpenn1**
https://rpubs.com/spenn1/sapflowqaqc

This script...
- Reads in sapflow data in the `Current data` folder
- Produces summary statistics
- Performs QA/QC tests: daily cycle, out of range, missing data, and trends
- Creates an interactive spacial map of quality flags
- Produces a summary table with number of flags per sapflow sensor

NOTE: Meeting w @aestearns01 this Thursday for a final lookthrough. **DO NOT MERGE YET**

Closes #60

### CODE REVIEW

**Comment by bpbond** [comment]
Looks good overall! A few things to consider:

* The 3.1-3.4 graphs feel very cramped; might make them bigger?
* 4.1 says "Daily test failure count" but are the graphs really showing counts? The legend is a bit confusing too.
* Also 4.1 graphs, use `coord_fixed()` to give them correct aspect ratio, i.e. corresponding to the real-world plot?

(Getting to this now, instead of post vacation, so you and Alice can consider.)

File: R_scripts/sapflow.Rmd [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+library(tidyr)
+library(ggplot2)
+library(plotly)
+theme_set(theme_minimal())
+
+# Global chunk options
+knitr:: opts_chunk$set(echo = FALSE)
+
+# Function setup
+read_sapflow <- function(filename) {
```
Comment (bpbond) [inline R_scripts/sapflow.Rmd:45]: Do you want to move this to a support file, as you did with the TEROS read function?

File: R_scripts/sapflow.Rmd [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    mutate(Deep_Sensor = grepl("D", Tree_Code),
+           Grid_Letter = substring(Grid_Square, 1, 1), 
+           Grid_Number = substring(Grid_Square, 2, 2)) -> sapflow
+
+  nomatch_ports <- anti_join(sf_raw, sf_inventory, by = c("Logger", "Port"))
+  if(nrow(nomatch_ports) > 0) {
+    warning("There were logger/port combinations that I couldn't find in sapflow_inventory.csv:")
+    nomatch_ports %>% 
+      distinct(Logger, Port) %>% 
+      print()
```
Comment (bpbond) [inline R_scripts/sapflow.Rmd:113]: Make this a `kable` so it looks nicer?

File: R_scripts/sapflow.Rmd [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+  bind_rows(.id = "test") %>% 
+  left_join(sf_inventory, by = c("Tree_Code", "Port")) %>% 
+  select(test, Plot, Tree_Code, Species, Port, 
+         Grid_Square, Logger, Installation_Date, failures) %>% 
+  mutate(Grid_Letter = substring(Grid_Square, 1, 1), 
+         Grid_Number = substring(Grid_Square, 2, 2)) -> daily_test
+
+# Spatial map of errors
+ggplot(filter(daily_test, Plot %in% c("C", "F", "S")), aes(x = Grid_Letter, y = Grid_Number)) + 
+  geom_jitter(aes(color = test)) + facet_wrap(~Plot, ncol = 1) + theme_minimal() -> p
```
Comment (bpbond) [inline R_scripts/sapflow.Rmd:456]: If using `geom_jitter()` I would do `set.seed(...)` at the beginning of the script. That way the points won't randomly jump around when you re-run script with exact same data.

**Review by bpbond — CHANGES_REQUESTED** [review]

File: R_scripts/sapflow.Rmd (hunk 1 above)
Comment (stephpenn1) [reply R_scripts/sapflow.Rmd]: Done!

**Review by bpbond — APPROVED** [review]

**Comment by bpbond** [comment]
@stephpenn1 This PR's check is failing. Let's schedule some time to understand why, and see how to fix. (Or, feel free to dive in without me of course!)

**Comment by stephpenn1** [comment]
I believe it has to do with what sapflow datasets are on `main`

**Comment by bpbond** [comment]
No, the checks are run on this branch. I can just fix, but thought you might be interested in some practice debugging GA.

**Comment by stephpenn1** [comment]
Ah, ok looks like a simple dependency issue, I can fix it!

**Comment by stephpenn1** [comment]
Ugh yeah lets talk abt this next week

**Comment by stephpenn1** [comment]
Looks like you fixed the issue @bpbond, ok to merge now?

**Comment by bpbond** [comment]
@stephpenn1 Sure. You absolutely diagnosed it correctly above 👍

**Pull request merged**